frontend: always enable qr-scanner when sending with lightning

thisconnect opened the pull request:
As it is very comon to scan an invoice, the QrScanner should already be enabled on the lightning send view.

thisconnect committed:
frontend: always enable qr-scanner when sending with lightning

As it is very comon to scan an invoice, the QrScanner should already
be enabled on the lightning send view.

thisconnect committed:
frontend: update lightning placeholders icons with the bolt icon

dangeross commented on frontends/web/src/routes/lightning/send/send.tsx:
```diff
@@ -64,13 +64,15 @@ export function Send({ accounts, code }: Props) {
-  const onChangeActiveScanQr = (active: boolean) => {
-    setActiveScanQr(active);
+  const onInvoice = (invoice: string) => {
+    // TODO: ensure this is a valid invoide?
```
`parseInput` calls `getParseInput` which validates the rawInput. It will also determine if its a bolt11 invoice or lnurl-pay

thisconnect replied:
then I remove this if  on the next line, that with `setRawInputError('is not a valid invoice');` ?

dangeross replied:
Does `ScanQRVideo` ever not return a string?

thisconnect replied:
It's always a string (coming from qr-scanner onResult callback), I remove the check.

thisconnect replied:
fixed

thisconnect committed:
frontend: use view for all send steps

Change to use a View for each step separatelly so that we can use
ViewButtons container or also ViewHeader in the future.

ViewButtons should position the bottons

thisconnect committed:
frontend: use success icon from view component

Changed to use ViewContent's built in animated success icon.

dangeross commented on frontends/web/src/routes/lightning/send/send.tsx:
```diff
@@ -64,13 +63,15 @@ export function Send({ accounts, code }: Props) {
+      setRawInputError('is not a valid invoice');
+      return;
+    }
+    setRawInputError(undefined);
```
I would suggest moving this line into the `parseInput` function to stop the flickering, maybe around L83

thisconnect replied:
Moved, but it still flickers quickly. Looks like it is coming from:

```
err Error: SdkError: Generic: Breez SDK error: Unrecognized input type
    at getApiResponse (lightning.ts:695:11)
    at async send.tsx:75:22
```

The reason was the the rawInput was first stored in a state, and then later (after the state update was done) it calls parseInput. I removed storing the invoice in a state and call parseInput directly with the invoice now.

dangeross commented on frontends/web/src/routes/lightning/send/send.tsx:
```diff
@@ -158,53 +159,59 @@ export function Send({ accounts, code }: Props) {
+                  <Status type="warning">{rawInputError}</Status>
+                )}
+                <ScanQRVideo onResult={onInvoice} />
+                {/* Note: unfortunatelly we probably can't read from HTML5 clipboard api directly in Qt/Andoird WebView */}
```
Seems so. Maybe an Input or Dialog to paste the clipboard into?

thisconnect replied:
Input field would work ofc, but  I think we want it as a button. Discussed with jokoono (not sure what his GH is), and left the button doing-nothing as a placeholder.

thisconnect committed:
frontend: improve loading and global error

Added loading icon while account, nodestate etc is syncing and a
error state in case something goes wrong.

thisconnect committed:
frontend: locate single back button on the left in ln send

A single back button in ViewButtons should be placed in reverse
order. This is the first time we use a back button only without
a promary action, as the UI automatically moves to the next step
when an invoice is detected.

Updated ViewButtons to accept reverseRow prop, to flip the order
in this case. This has no impoact on small screen as the buttons
are not rendered in a row, but in a column.

thisconnect committed:
frontend: minor styling update on lightning send confirm view

Small improvements, updated font-sizes and spacings.

thisconnect committed:
frontend: simplify ilghtning send states

This fixes the error component quickly flickering up on the
select-invoice view.

Removed some state that is not needed, also passing the invoice
directly to parseInput removes the error flickering.

thisconnect committed:
frontend: keep paste button for now

Lets keep the paste button even though it is not yet implemented,
so that we don't forget.

thisconnect committed:
frontend: add sending step to lightning send view

Paying an invoice can take a moment, this commit shows a sending
waiting view after the user confirmed the payment.

dangeross reviewed (approved):
Looks good! Will merge and build
